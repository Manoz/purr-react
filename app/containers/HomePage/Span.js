import styled from 'styled-components';

const Span = styled.span`
  color: ${props => props.secondary ? 'var(--secondaryColor)' : '#555'};
`;

export default Span;
