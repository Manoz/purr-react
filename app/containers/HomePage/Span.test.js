// See guide here: https://github.com/styled-components/jest-styled-components
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Span from './Span';

describe('<Span />', () => {
  it('should match the snapshot', () => {
    const wrapper = renderer
      .create(<Span>test</Span>)
      .toJSON();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    const wrapper = renderer.create(<Span />).toJSON();
    expect(wrapper).toMatchSnapshot();
    expect(wrapper).toHaveStyleRule('color', '#555');
  });

  it('applies style according to passed props', () => {
    const wrapper = renderer.create(<Span secondary />).toJSON();
    expect(wrapper).toHaveStyleRule('color', 'var(--secondaryColor)');
  });
});
