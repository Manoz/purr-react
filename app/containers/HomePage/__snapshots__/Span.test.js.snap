// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Span /> should match the snapshot 1`] = `
.c0 {
  color: #555;
}

<span
  className="c0"
>
  test
</span>
`;

exports[`<Span /> should render without throwing an error 1`] = `
.c0 {
  color: #555;
}

<span
  className="c0"
/>
`;
